src/lib/layer.js

```diff
@@ -446,7 +446,10 @@ export default class Layer {
     settings.polygonOffset = offsets;
 
     // Call subclass lifecycle method
-    this.draw({moduleParameters, uniforms, settings});
+    withParameters(this.context.gl,
```

Comment:
The original idea was that we would forward the settings to the layer's draw function, but we can certainly apply them here. 

We wanted to support `props.settings`, this is a nice way to do it. Won't allow overriding the layers hardcoded draw settings though, but maybe that is a good thing.

Not sure if we are actually supplying `props.settings `somewhere, maybe they are already supplied by layer-manager or draw-and-pick?)